classes/movie/source.rb

```diff
@@ -0,0 +1,15 @@
+class Source
+  attr_accessor :name, :id
+  attr_reader :items
+
+  def initialize(name, id = nil)
+    @id = id || Random.rand(1..500)
+    @name = name
+    @items = []
+  end
+
+  def add_item(item)
+    @items << item
+    item.add_source(self)
```

Comment:
- You have done well so far, but after correctly implementing the custom setter method, this line of code should be item.source = self. Please fix this 🙏